micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MicronautProcessEngineConfiguration.java

```diff
@@ -29,10 +32,15 @@
 
     private final DatasourceConfiguration datasourceConfiguration;
 
-    public MicronautProcessEngineConfiguration(ApplicationContext applicationContext, Configuration configuration, DatasourceConfiguration datasourceConfiguration) {
+    private final List<Consumer<ProcessEngineConfiguration>> processEngineConfigurationCustomizers;
```

Comment:
I'm not quite sure if the `List` is good here or an `Optional` would be better. Both will work and the list allows multiple customizers which seems more generic. However, there is no fixed order (to my knowledge) so this might lead to problems if the execution is random.

Maybe the `Optional` is safer for now. Do you have a strong opinion?

Reply:
I am not sure if having an `Optional` is a good idea. The convention says that `Optional` should be used as a return value. It is not recommended to use it as a field.

One possible solution would be to declare a default `ProcessEngineConfigurationCustomizer` that does nothing and can be easily replaced by a Customizer provided by the developer.

Reply:
We can have either multiple customizers or only one. With multiple we can apply `OrderUtil.sort()` to the list to allow users to use `Ordered` on configuration customizers to fine tune their execution order. If we want only one customizer, the only option I know about is to use `Optional`, since this is Micronaut workaround for optional beans.

Reply:
Currently, I've added `OrderUtil.sort()` to customizers list. Please check it in the code and give me a feedback which approach you preffer (`Optional` or `List`)?

Reply:
@l0co, @carlosbarragan , thanks for the discussion.

@carlosbarragan suggested to have a default implementation. I'm not a Micronaut expert but I stumbled across https://docs.micronaut.io/latest/guide/index.html#replaces and I think `@Replaces` is what we should be using to replace the default bean.

I tried this and pushed it to https://github.com/tobiasschaefer/micronaut-camunda-bpm/tree/config-customizer The last commit contains my changes. Note: I also fixed my two review comments in there.

@l0co: Feel free to cherry-pick this into your branch and squash it.

If you agree then this could be merged to master.

Reply:
Hi @l0co,

I talked to @carlosbarragan and we would like to merge your pull request together with the last commit on https://github.com/tobiasschaefer/micronaut-camunda-bpm/tree/config-customizer by squashing them all with you as author.

Please have a look at the changes. Do you have any more comments or can we merge it also from your point of view?

Thanks
Tobias

Reply:
No. I think it's OK right now.